### PR TITLE
Improvement on votekick

### PR DESCRIPTION
**Pull request by frieda666**
- Allow admins to cancel a votekick
  `/vote c`
- Allow players to change their vote
  `/vote yes` after `/vote no` will override the previous vote
- Require a reason to start a votekick
  `/votekick player1 some reason`

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/src/mindustry/core/NetServer.java [hunk 1]
```diff
@@ -422,6 +426,8 @@ boolean checkPass(){
                     builder.append("[lightgray] ").append(p.name).append("[accent] (#").append(p.id()).append(")\n");
                 });
                 player.sendMessage(builder.toString());
+            }else if(args.length == 1){
+                player.sendMessage("[orange]You need a valid reason to kick the player. Add a reason after the player name.");
```
Comment (Anuken) [inline core/src/mindustry/core/NetServer.java:430]: This would break the votekick button in the player list.

Comment (frieda666) [reply core/src/mindustry/core/NetServer.java:430]: Fixed that.

**Comment by nanmenyangde** [comment]
when clicking the votekick button, why not display a input box for the player to type their kick reason?
@frieda666

**Comment by nanmenyangde** [comment]
if not this feature (require reason) may be almost useless, because players are used to clicking the button instead of using the command

**Comment by Anuken** [comment]
That was added in another PR @nanmenyangde

**Pull request merged**